packages/techdocs-common/src/helpers.ts

```diff
@@ -50,9 +60,113 @@ export const parseReferenceAnnotation = (
   };
 };
 
+/**
+ * Check if the entity provides a `backstage.io/techdocs-ref` annotation of type `dir`
+ * and return the annotation. It accepts the two variants `<target>` and `dir:<target>`.
+ *
+ * @param entity - the entity to check
+ */
+export const getDirLocation = (
+  entity: Entity,
+): { type: 'dir'; target: string } | undefined => {
+  const annotation = entity.metadata.annotations?.['backstage.io/techdocs-ref'];
+
+  if (!annotation) {
+    return undefined;
+  }
+
+  if (typeof annotation !== 'string') {
+    return undefined;
+  }
+
+  // if the string doesn't contain `:`, interpret it as the target of a dir type
+  if (!annotation.includes(':')) {
```

Comment:
Hey @dhenneke, @ottosichert @camilaibs and I were taking a look at this and we're wondering if there's any utility in allowing a non-prefixed ref value beyond 4-5 saved characters?

Reply:
Yes you can boil it down to just saving 4 characters. Considering that we can probably expect that `backstage.io/techdocs-ref: .` will be the default for the majority of use-cases, I find it _better_ from a user perspective. We should also consider that `<type>:<target>` is well-known across Backstage as it is also used in the catalog. **But**, a normal Backstage user will probably never need to understand that there are different types of locations where an entity might origin. They register their entities via the catalog-import or get it registered by the scaffolder, and don't need to write any location reference by hand. Even if they choose to use a [`Location` entity](https://backstage.io/docs/features/software-catalog/descriptor-format#kind-location), it has code to resolve [either relative or absolute locations](https://github.com/backstage/backstage/blob/20bb197b6f7fc4e7b1227b1c0ab7e47e90a4ca6e/plugins/catalog-backend/src/ingestion/processors/LocationEntityProcessor.ts#L23-L39) – it even doesn't support `<type>:<location>` at all. But they will need to do it for the `backstage.io/techdocs-ref` annotation and from this point-of-view, I would argue that it is more consistent if it also supports the short-form.

But there is also another perspective of this and this is "documentation" (i.e. of the feature). There are still some places that prefer `<type>:<target>` and have no automatic logic for a short form. However, this is primarily relevant for integrators. For them, it might be easier to understand `dir:.` (even though the catalog only supports `file`), and that this also implies that there might be other alternatives (-> `url`). [TechDocs - Creating and publishing your docs](https://backstage.io/docs/features/techdocs/creating-and-publishing) might be easier to read if it just lists (a) `dir` (recommended): relative location based on your entity definition file, and (b) `url` (advanced): absolute url to a location (e.g. a GitHub/GitLab/... repository). Compared to this, an additional paragraph "you can also skip the `dir:` if you like, but please never skip the `url:` since this will break your entity" might introduce confusion that might result in issues or support questions. A possible outcome of this argument could be to either skip the "short-form" or to make it compatible with URLs too.

Long story short: I would like to keep this feature but don't plan to add support for plain Urls. But I would also be fine with removing it and just keep the updated `dir:` behavior and the establishment of a new standard.

Reply:
> _But, a normal Backstage user will probably never need to understand that there are different types of locations where an entity might originate. They register their entities via the catalog-import or get it registered by the scaffolder, and don't need to write any location reference by hand._

Ah-hah.  Yeah, this resonates.  Backstage is probably overly guilty of this more broadly (leakage of knowledge/expectations between persona boundaries)...  So I like the idea of simplifying.

Reply:
Although allowing a short-hand notation seems like a minor aspect, we have taken the time to go through all implications of this, i.e.:

Pros
- Less to type, looks clean
- Backstage/TechDocs users don't need to be aware of `<type>:<target>` notation

Cons
- `url:https://example.com` is a valid (non-windows) folder name
- Two ways to specify directories: `./path` or `dir:./path` might cause confusion
- Implementation is more complex due to discovery of short-hand

Also, in the future the requirement of the ref might be completely dropped by using catalog processors, thus being more explicit at this point would be the best option to move forward in our point of view.

Reply:
What do you think @dhenneke?

Reply:
@ottosichert I think these are valid arguments and I agree that it will be best to remove the special handling of the short-hand notation. Should only be a few places and tests as the majority of code is concerning the actual `dir:` location. I'll update the PR.